src/Tags.zig

```diff
@@ -44,17 +44,23 @@ pub fn findTags(self: *Tags, fname: []const u8) anyerror!void {
         var file = try std.fs.cwd().openFile(fname, .{});
         defer file.close();
 
-        const size = try file.getEndPos();
+        const metadata = try file.metadata();
+        const size = metadata.size();
```

Comment:
Why is this better than `getEndPos()`?

Reply:
Not much. Both call `fstat` to get information about opened file. I use it to get file type.

When I first use ztags, I pass `src` directory to it, it throw following errors
```
thread 32397559 panic: reached unreachable code
~/.versions/zig-macos-aarch64-0.11.0-dev.669+bd5a8f86a/lib/std/os.zig:4305:19: 0x10270cc47 in mmap (ztags)
        .INVAL => unreachable, // Invalid parameters to mmap()
                  ^
```

Error message above doesn't make much sense, I have no clue which parameter is wrong. After my patch, it will report 

```
error: NotFile
~/ztags/src/Tags.zig:54:13: 0x1000e26a7 in findTags (ztags)
            return error.NotFile;
            ^
```

Reply:
Let's `catch` the `NotFile` error in `main.zig` and print a human friendly error message instead of the error trace.

```zig
tags.findTags(full_fname) catch |err| switch (err) {
    err.NotFile => {
        try std.io.getStderr().writer().print(
            "Error: {} is a directory. Arguments must be Zig source files.\n",
            .{full_fname},
        );
        return 22; // EINVAL
    },
    else => return err,
};
```

Reply:
Done.